### PR TITLE
Initial version of CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by Aayush-Kasurde**
Fixes documentation issue #70 

Signed-off-by: Aayush Kasurde <aayush.kasurde@gmail.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+Guidelines for bug reports:
+
+1. **Use the GitHub issue search** &mdash; check if the issue has already been
+   reported.
+
+2. **Check if the issue has been fixed** &mdash; try to reproduce it using the
+   latest `master` or development branch in the repository.
+
+3. **Isolate the problem** &mdash; ideally create a [reduced test
+   case](http://css-tricks.com/6263-reduced-test-cases/) and a live example.
```
Comment (temoto) [inline CONTRIBUTING.md]: css-tricks links to 404

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+## Using the issue tracker
+
+The [issue tracker](https://github.com/eventlet/eventlet/issues) is
+the preferred channel for [bug reports](#bugs), [features requests](#features)
+and [submitting pull requests](#pull-requests), but please respect the following
+restrictions:
+
+* Please **do not** use the issue tracker for personal support requests (use
+  [Stack Overflow](http://stackoverflow.com) or IRC).
```
Comment (temoto) [inline CONTRIBUTING.md]: There's no such rule in Eventlet. Questions are fine. No trolling is common sense, so redundant also.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+<a name="bugs"></a>
+## Bug reports
+
+A bug is a _demonstrable problem_ that is caused by the code in the repository.
+Good bug reports are extremely helpful - thank you!
+
+Guidelines for bug reports:
+
+1. **Use the GitHub issue search** &mdash; check if the issue has already been
```
Comment (temoto) [inline CONTRIBUTING.md]: Recommend Google search, there are too many places where good Eventlet related information can be found.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+2. **Check if the issue has been fixed** &mdash; try to reproduce it using the
+   latest `master` or development branch in the repository.
+
+3. **Isolate the problem** &mdash; ideally create a [reduced test
+   case](http://css-tricks.com/6263-reduced-test-cases/) and a live example.
+
+A good bug report shouldn't leave others needing to chase you up for more
+information. Please try to be as detailed as possible in your report. What is
+your environment? What steps will reproduce the issue? What version of Nginx
```
Comment (temoto) [inline CONTRIBUTING.md]: -Nginx
+uname -a, python -V, pip freeze

**Comment by Aayush-Kasurde** [comment]
@temoto All comments are accepted. I will work on review-comment and update this PR

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+experiences the problem? What would you expect to be the outcome? All these
+details will help people to fix any potential bugs.
+
+Example:
+
+> Short and descriptive example bug report title
```
Comment (temoto) [inline CONTRIBUTING.md]: Example is something to follow. "Short and descriptive example bug report title" is a recipe, not an example, so there's contradiction in itself. :-)

Examples of good bug report title:
`eventlet.hubs.trampoline throws socket.timeout exception without 'timed out' argument`
`PyPy + eventlet + urllib3: ValueError: file descriptor cannot be a negative integer (-1)`

examples of less useful title:
`Test errors`
`eventlet hanging`

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+>
+> A summary of the issue and server version / OS / browser environment in which
+> it occurs. If suitable, include the steps required to reproduce the bug.
+>
+> 1. This is the first step
+> 2. This is the second step
+> 3. Further steps, etc.
```
Comment (temoto) [inline CONTRIBUTING.md]: And it goes on not providing a real example to follow. Here's a recent ultimately useful bug report to get example from: https://github.com/eventlet/eventlet/issues/226

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+> causing the bug, and potential solutions (and your opinions on their merits).
+
+
+<a name="features"></a>
+## Feature requests
+
+Feature requests are welcome. But take a moment to find out whether your idea
+fits with the scope and aims of the project. It's up to *you* to make a strong
+case to convince the project's developers of the merits of this feature. Please
+provide as much detail and context as possible.
```
Comment (temoto) [inline CONTRIBUTING.md]: Again, common sense, a whole section without useful information.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+
+3. Create a new topic branch (off the main project development branch) to
+   contain your feature, change, or fix:
+
+   ```bash
+   git checkout -b <topic-branch-name>
+   ```
+
+4. Commit your changes in logical chunks. Please adhere to these [git commit
+   message guidelines](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)
```
Comment (temoto) [inline CONTRIBUTING.md]: Again, link to a page full of text about appearance of commit, saying nothing about their content except for "logical chunks" which is not quite to Eventlet pull request policy, found on http://eventlet.net/ index page.

"or your code is unlikely to be merged" is far from truth. I often squash several commits and rephrase text if the author doesn't mind. And almost all contributors only care for their patch to be merged, no matter how commit looks like. And that's very fine.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+   or your code is unlikely be merged into the main project. Use Git's
+   [interactive rebase](https://help.github.com/articles/interactive-rebase)
+   feature to tidy up your commits before making them public.
+
+5. Locally merge (or rebase) the upstream development branch into your topic branch:
```
Comment (temoto) [inline CONTRIBUTING.md]: rebase. Eventlet contains no merges since 2012 and I'm proud of simple linear history.

**Pull request closed without merging**